File: angular-spring-boot/backend/src/main/java/ru/isnagornov/templates/service/AbstractService.java
package ru.isnagornov.templates.service;

import ru.isnagornov.templates.mapper.CommonMapper;

import java.util.List;

public abstract class AbstractService<T, M extends CommonMapper<T>> {

    protected M mapper;

    public AbstractService(M mapper) {
        this.mapper = mapper;
    }

    public void add(T entity) {
        this.mapper.add(entity);
    }

    public T getById(Long id) {
        return this.mapper.getById(id);
    }

    public List<T> getAll() {
        return this.mapper.getAll();
    }

    public void update(T entity) {
        this.mapper.update(entity);
    }

    public void delete(Long id) {
        this.mapper.delete(id);
    }
}

File: angular-spring-boot-kotlin/backend/src/main/kotlin/ru/isnagornov/templates/service/AbstractService.kt
package ru.isnagornov.templates.service

import ru.isnagornov.templates.mapper.CommonMapper

abstract class AbstractService<T, M : CommonMapper<T>>(protected var mapper: M) {

    val all: List<T>
        get() = this.mapper.all

    fun add(entity: T) {
        this.mapper.add(entity)
    }

    fun getById(id: Long?): T {
        return this.mapper.getById(id)
    }

    fun update(entity: T) {
        this.mapper.update(entity)
    }

    fun delete(id: Long?) {
        this.mapper.delete(id)
    }
}
